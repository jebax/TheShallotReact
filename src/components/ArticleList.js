import React, { Component } from 'react'
import axios from 'axios'
import { config } from '../config.js'
import ArticlePreview from './ArticlePreview'

export default class ArticleList extends Component {
  state = {
    articles: []
  }

  componentDidMount() {
    axios.get(config.guardianUrl).then(response => {
      console.log(response)
      this.setState(prevState => ({
        articles: response.data.response.results
      }))
    })
  }

  displayPreview = (article, index) => {
    return (
      <article
        key={index}
      >
        <ArticlePreview
          id={index}
          imageUrl={article.fields.thumbnail}
          headline={article.webTitle}
          url={article.webUrl}
        />
      </article>
    )
  }

  render() {
    return (
      <div
        id='article-list'
      >
        {this.state.articles.map((article, index) => {
          return this.displayPreview(article, index)
        })}
      </div>
    )
  }
}
