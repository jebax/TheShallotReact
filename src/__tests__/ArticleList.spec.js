import React from 'react'
import axios from 'axios'
import { shallow } from 'enzyme'
import ArticleList from '../components/ArticleList'

jest.mock('axios')
jest.useFakeTimers()

describe('<ArticleList />', () => {
  var wrapper

  beforeAll(() => {
    let response = {}
    axios.get.mockResolvedValue(response)
    wrapper = shallow(<ArticleList />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the article list wrapper', () => {
    expect(wrapper.find('#article-list').length).toEqual(1)
  })

  it('fetches information from the correct API url when mounted', () => {
    let response = {}
    axios.get.mockResolvedValue(response)

    const secondWrapper = shallow(<ArticleList />)
    expect(axios.get).toHaveBeenCalledTimes(2)
  })
})
