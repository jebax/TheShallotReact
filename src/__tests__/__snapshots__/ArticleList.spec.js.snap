// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleList /> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article>
          <ArticlePreview
            headline="Title1"
            id={0}
            image="Img1"
            url="Url1"
          />
        </article>,
        <article>
          <ArticlePreview
            headline="Title2"
            id={1}
            image="Img2"
            url="Url2"
          />
        </article>,
        <article>
          <ArticlePreview
            headline="Title3"
            id={2}
            image="Img3"
            url="Url3"
          />
        </article>,
      ],
      "id": "article-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <ArticlePreview
            headline="Title1"
            id={0}
            image="Img1"
            url="Url1"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "headline": "Title1",
            "id": 0,
            "image": "Img1",
            "url": "Url1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "article",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <ArticlePreview
            headline="Title2"
            id={1}
            image="Img2"
            url="Url2"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "headline": "Title2",
            "id": 1,
            "image": "Img2",
            "url": "Url2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "article",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <ArticlePreview
            headline="Title3"
            id={2}
            image="Img3"
            url="Url3"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "headline": "Title3",
            "id": 2,
            "image": "Img3",
            "url": "Url3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "article",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article>
            <ArticlePreview
              headline="Title1"
              id={0}
              image="Img1"
              url="Url1"
            />
          </article>,
          <article>
            <ArticlePreview
              headline="Title2"
              id={1}
              image="Img2"
              url="Url2"
            />
          </article>,
          <article>
            <ArticlePreview
              headline="Title3"
              id={2}
              image="Img3"
              url="Url3"
            />
          </article>,
        ],
        "id": "article-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <ArticlePreview
              headline="Title1"
              id={0}
              image="Img1"
              url="Url1"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "headline": "Title1",
              "id": 0,
              "image": "Img1",
              "url": "Url1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "article",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <ArticlePreview
              headline="Title2"
              id={1}
              image="Img2"
              url="Url2"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "headline": "Title2",
              "id": 1,
              "image": "Img2",
              "url": "Url2",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "article",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <ArticlePreview
              headline="Title3"
              id={2}
              image="Img3"
              url="Url3"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "headline": "Title3",
              "id": 2,
              "image": "Img3",
              "url": "Url3",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "article",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
